components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceManagementServiceImpl.java

```diff
@@ -471,6 +472,33 @@ public Response getDeviceLocation(
 
     }
 
+
+    @GET
+    @Path("/{type}/{id}/info")
+    @Override
+    public Response getDeviceInformation(
+            @PathParam("type") @Size(max = 45) String type,
+            @PathParam("id") @Size(max = 45) String id,
+            @HeaderParam("If-Modified-Since") String ifModifiedSince) {
```

Comment:
Is this parameter being used?

Reply:
This is a requirement from some customers, we have not used it an of now. It should be used with caching.

Reply:
If the customer sets this value, without internal details they would assume we are sending modified values only.  This is a public API from product.

So how about throwing an Unsupported exception if this is set to let know customer we still haven't implemented this.
Or else, we can easily overload this method later and only support type and id for now.

Reply:
This value is included in most the resources, So that is why I also added it. We should implement it for other resources as well.